### DISCUSSION TITLE
flatten PDF documents with annotations?

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#474] [Q&A] [answered]
@pgundlach,

I wonder whether this might be doable with LuaTeX.

This [sample PDF document](https://www.aloaha.com/wp-content/uploads/2016/07/Beschwerde_wegen_Falschauskunft.pdf) contains a number of annotations which are lost in the resulting `<Image>`.

Original page (printed in grayscale):

![afes-000001-fs8](https://user-images.githubusercontent.com/51094960/224297991-ea02b4dd-6f01-47bf-8b97-445eac2dfa98.png)

Imported `<Image>`:

![aafes-000001-fs8](https://user-images.githubusercontent.com/51094960/224297985-3de429ac-8d30-462a-a6a9-fa89de5547d2.png)

I guess that LuaTeX cannot import PDF annotations as such, but I wonder whether it could flatten them before `<Image>` (as if they were printed with a virtual printer).

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
You mean like removing the annotations? This should be doable with LuaTeX, but likely not trivial.

**Reply by pr-apes** [reply]
I thought it was easier to apply the normal appearance stream to annotations (in order to convert them to standard content).
